multibuildingdetector/__main__.py

```diff
@@ -36,12 +36,16 @@ def run(input, output, batch_size, iterator='SerialIterator', device=-1,
         chainer.cuda.get_device_from_id(device).use()
         model.to_gpu()
 
-    train = TransformDataset(
-        xmldataset.XMLDataset(input),
+    dataset = xmldataset.XMLDataset(input)
```

Comment:
We should not implement the train test split logic in `__main__.py`. The dataset should handle this.

Reply:
The dataset class should only contain and serve data in the format we want it to.
Everything else, like what it is used for, should be handled separately, like train/test split before initializing the iterators and start training.